src/tools/perf/lib/libperf.c

```diff
@@ -985,6 +985,110 @@ static void ucp_perf_test_err_handler(void *arg, ucp_ep_h ep,
               ucs_status_string(status));
 }
 
+static ucs_status_t ucp_perf_test_loopback_ep_create(ucx_perf_context_t *perf,
```

Comment:
the flag is necessary, ctx get group size much later

+    if (UCX_PERF_TEST_FLAG_LOOPBACK_CONN & ctx->params.super.flags) {
+        ctx->flags |= (TEST_FLAG_PRINT_TEST |
+                       TEST_FLAG_PRINT_RESULTS);
+
+        ctx->sock_rte_group.size      =  1;

Reply:
question is why it's needed earlier?
maybe it's needed only for perftest.c to track whether -l flag is passed, then it's ok
but for the "libperf.h" API, I think it's not really needed. libperf can use group_size to know if it's loopback or not

Reply:
because group_size sits on the ctx - ctx init stage is later than params parsing... in additional it's perf test already uses this approach for other cmd params - case 'r' for example or case 'o'.... another reason - gtests I've added, this is a little bit issue to do it without the flag

Reply:
really don't understand what's the issue with the special flag... for the person who reads the code isn't clear group size 1 <=> loopback... it could be two diff eps on the same procs for example